sorc/IMSaggregate_mod.f90

```diff
@@ -333,13 +338,38 @@ subroutine write_IMS_outputs_vec(idim, jdim, otype, date_str,scfIMS, sndIMS, lon
     nobs = count (abs(scfIMS -nodata_real) > nodata_tol)
     print *, 'writing out', nobs, ' observations'
 
+    if(date_from_file == "ascifile") then
+      time_char =  date_str(1:8)//date_str(10:11)
+    endif
+
     allocate(data_vec(2,nobs)) 
     allocate(coor_vec(3,nobs)) 
 
-    !--- define dimension
+    !--- define spatial dimension
     error = nf90_def_dim(ncid, 'numobs', nobs, id_obs)
     call netcdf_err(error, 'defining obs dimension' )
- 
+
+    ! --- define global attributes and units for valid_time_str and valid_epoch_time
+    if(date_from_file == "ascifile") then
+
+        error = nf90_put_att(ncid, NF90_GLOBAL, "valid_time_str", time_char)
+        call netcdf_err(error, 'put valid_date_str as global attribute')
+
+        error = nf90_put_att(ncid, NF90_GLOBAL, "comment", "This is YYYYMMDDHH read from ascii file.")
```

Comment:
Last comment, then I think this is ready, make this `valid_time_str_comment` to match the other attribute below

Reply:
Now it is below:
netcdf IMSscf.20191201.C96 {
dimensions:
        numobs = 12252 ;
variables:
        double lon(numobs) ;
                lon:long_name = "longitude" ;
        double lat(numobs) ;
                lat:long_name = "latitude" ;
        double oro(numobs) ;
                oro:long_name = "orography" ;
        double IMSscf(numobs) ;
                IMSscf:long_name = "IMS snow covered fraction" ;
                IMSscf:units = "-" ;
        double IMSsnd(numobs) ;
                IMSsnd:long_name = "IMS snow depth" ;
                IMSsnd:units = "mm" ;

// global attributes:
                :valid_time_str = "2019120100" ;
                :**valid_time_str_comment = "This is the 00Z reference time. Note that products are nowcasted to be valid specifically at the time given here."** ;
}